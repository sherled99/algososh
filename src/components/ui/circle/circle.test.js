import React from "react";
import renderer from "react-test-renderer";
import { Circle } from "./circle"; // Путь к вашему компоненту
import { ElementStates } from "../../../types/element-states";

describe("Circle Component Snapshots", () => {
  it("renders correctly without letter and elements", () => {
    const tree = renderer.create(<Circle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with a letter", () => {
    const tree = renderer.create(<Circle letter="A" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with a head", () => {
    const tree = renderer.create(<Circle head="Head" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with a tail", () => {
    const tree = renderer.create(<Circle tail="Tail" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with an index", () => {
    const tree = renderer.create(<Circle index={1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly as small circle", () => {
    const tree = renderer.create(<Circle isSmall />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly in default state", () => {
    const tree = renderer.create(<Circle state={ElementStates.Default} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly in changing state", () => {
    const tree = renderer.create(<Circle state={ElementStates.Changing} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly in modified state", () => {
    const tree = renderer.create(<Circle state={ElementStates.Modified} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with a react element in head", () => {
    const reactElement = <span>React Element in Head</span>;
    const tree = renderer.create(<Circle head={reactElement} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with a react element in tail", () => {
    const reactElement = <span>React Element in Tail</span>;
    const tree = renderer.create(<Circle tail={reactElement} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
