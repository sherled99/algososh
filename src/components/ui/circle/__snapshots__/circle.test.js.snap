// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle Component Snapshots renders correctly as small circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
    data-testid="default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Component Snapshots renders correctly in changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
    data-testid="changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Component Snapshots renders correctly in default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Component Snapshots renders correctly in modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
    data-testid="modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Component Snapshots renders correctly with a head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    Head
  </div>
  <div
    className="circle   default"
    data-testid="default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Component Snapshots renders correctly with a letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Component Snapshots renders correctly with a react element in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <span>
      React Element in Head
    </span>
  </div>
  <div
    className="circle   default"
    data-testid="default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Component Snapshots renders correctly with a react element in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <span>
      React Element in Tail
    </span>
  </div>
</div>
`;

exports[`Circle Component Snapshots renders correctly with a tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    Tail
  </div>
</div>
`;

exports[`Circle Component Snapshots renders correctly with an index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle Component Snapshots renders correctly without letter and elements 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="default"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
